### ISSUE TITLE
Some links in the online documentation are broken

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3177]
|<img alt="mim's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| mim reported|
|-|-|
|Trac ID|trac#3177|
|Type|defect|
|Created|2008-04-11 16:29:26Z|
|Branch|https://github.com/twisted/twisted/tree/api-links-broken-3177|

For example in the following page most of the links to howto documents are broken: http://twistedmatrix.com/projects/core/documentation/howto/plugin.html 

i.e.

http://twistedmatrix.com/projects/core/documentation/howto/twisted.plugin
http://twistedmatrix.com/projects/core/documentation/howto/zope.interface
http://twistedmatrix.com/projects/core/documentation/howto/twisted.plugin.getPlugins
http://twistedmatrix.com/projects/core/documentation/howto/IPlugin

all return 404

<details><summary>Searchable metadata</summary>

```
trac-id__3177 3177
type__defect defect
reporter__mim mim
priority__highest highest
milestone__ 
branch__branches_api_links_broken_3177 branches/api-links-broken-3177
branch_author__ 
status__closed closed
resolution__fixed fixed
component__website website
keywords__documentation documentation
time__1207931366000000 1207931366000000
changetime__1220228091000000 1220228091000000
version__None None
owner__thijs thijs
cc__thijs cc__exarkun
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph commented|
|-|-|

I could have sworn this was already reported, but I can't find the duplicate.  Updating the summary to be more descriptive instead.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_2"></a>|@radix commented|
|-|-|

You may have been thinking of [#3130](https://github.com/twisted/twisted/issues/3130), which represents the code fix necessary to fix this issue.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

[#3227](https://github.com/twisted/twisted/issues/3227) was a duplicate of this.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra commented|
|-|-|

What is the status of this ticket? The links are still broken here. How can we help?

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra commented|
|-|-|

[#7125](https://github.com/twisted/twisted/issues/7125) is related to this and probably a duplicate of this one (or vice versa).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun commented|
|-|-|

There are three bugs described by this ticket.

  1. The initial API link to twisted.plugin used to be invalid, but no longer is, due to a mistaken in how the API documentation was generated.  This part of the ticket is already resolved.
  1. The document contains API links to `zope.interface`.  Lore does not support multi-project API link generation.  The fix for this in the short term is probably to remove the API class from that link.  In the long term, we might add this feature to Lore.
  1. The later API links which are broken are misspecified in the document and just need to be fixed by being made fully qualified or by the addition of a base attribute to the code tags.

The latter two should be very simple to resolve and the former is already fixed.  [#7125](https://github.com/twisted/twisted/issues/7125) isn't really a duplicate, as it is about linking to howto documentation in different subprojects which the plugin howto does not do.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_10"></a>|@thijstriemstra commented|
|-|-|

(In [24643]) Branching to 'api-links-broken-3177'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_11"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

Replying to [exarkun](#note_8):
> The latter two should be very simple to resolve and the former is already fixed.
>
Think they're resolved in r24644, putting it up for review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_12"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

I don't think there should be whitespace between `<code>` and `IMaterial`.  It's probably harmless, but sometimes it can be a problem (it is in `class="API"` tags, at least).  So better to just be consistent about whitespace everywhere, I think, even though it makes the paragraph margins ugly. :)

Other than that, looks good, please merge after the very minor whitespace issue is resolved.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_14"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [24654]) Merge api-links-broken-3177: fix some broken api links in the plugin howto.

Author: thijs
Reviewer: exarkun
Fixes: [#3177](https://github.com/twisted/twisted/issues/3177)
